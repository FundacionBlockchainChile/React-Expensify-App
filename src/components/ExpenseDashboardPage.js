import React from "react";
import ExpenseList from "./ExpenseList";
import ExpenseListFilters from './ExpenseListFilters'

const ExpeneDashboardPage = () => (
  <div>
    <ExpenseListFilters />
    <ExpenseList />
  </div>
);

export default ExpeneDashboardPage;