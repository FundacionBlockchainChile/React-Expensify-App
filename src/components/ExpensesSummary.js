import React from "react";
import { connect } from "react-redux";
import selectExpenses from "../selectors/expenses";
import expensesTotal from "../selectors/expenses-total";
import numeral from "numeral";

export const ExpensesSummary = (props) => (
         <div>
           <h3>Viewing {props.expenses.length} expenses totalling {numeral(props.total / 100).format('$0,0.00')}</h3>
         </div>
);

const mapStateToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters),
    total: expensesTotal(selectExpenses(state.expenses, state.filters))
  };
};

export default connect(mapStateToProps)(ExpensesSummary);