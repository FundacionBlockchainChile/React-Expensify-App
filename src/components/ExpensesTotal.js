import React from "react";
import { connect } from "react-redux";
import selectExpenses from "../selectors/expenses";

export const ExpenseTotal = (props) => (
         <div>
           <h3>You have {props.expenses.length} number of expenses.</h3>
         </div>
);

const mapStateToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters),
  };
};

export default connect(mapStateToProps)(ExpenseTotal);