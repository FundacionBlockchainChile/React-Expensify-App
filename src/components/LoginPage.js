import React from "react";
import { connect } from 'react-redux'
import { startLogin, startLoggin } from '../actions/auth'
import { NavLink } from "react-router-dom";

export const LoginPage = ({ startLogin }) => (
         <div>
           <h1>Welcome</h1>
           <button onClick={startLogin}>Login</button>
         </div>
       );

const mapDispatchToProps = (dispatch) => ({
  startLogin: () => dispatch(startLogin()),
});

export default connect (undefined, mapDispatchToProps)(LoginPage)