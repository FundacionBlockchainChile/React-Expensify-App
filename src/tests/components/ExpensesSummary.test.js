import React from "react";
import { shallow } from "enzyme";
import { ExpensesSummary } from "../../components/ExpensesSummary";
import expenses from "../fixtures/expenses";


test("should render ExpenseSummary with 1 expense", () => {
  const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={235} />);
  expect(wrapper).toMatchSnapshot();
});

test("should render ExpenseSummary with number total", () => {
  const wrapper = shallow(<ExpensesSummary expensesCount={23} expensesTotal={23532525234542} />);
  expect(wrapper).toMatchSnapshot();
});
