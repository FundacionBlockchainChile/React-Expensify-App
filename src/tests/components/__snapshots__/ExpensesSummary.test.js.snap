// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseSummary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling
     
    $2.35
  </h1>
</div>
`;

exports[`should render ExpenseSummary with number total 1`] = `
<div>
  <h1>
    Viewing 
    23
     
    expenses
     totalling
     
    $235,325,252,345.42
  </h1>
</div>
`;
